sys/src/main/java/com/alibaba/nacos/sys/env/NacosAutoRefreshPropertySourceLoader.java

```diff
@@ -0,0 +1,332 @@
+import com.alibaba.nacos.sys.file.WatchFileCenter;
+import com.alibaba.nacos.sys.utils.ApplicationUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.boot.env.OriginTrackedMapPropertySource;
+import org.springframework.boot.env.PropertySourceLoader;
+import org.springframework.boot.origin.Origin;
+import org.springframework.boot.origin.OriginTrackedValue;
+import org.springframework.boot.origin.TextResourceOrigin;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.io.Resource;
+import org.springframework.util.Assert;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.LineNumberReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Support for configuring automatic refresh and loading into the Environment.
+ *
+ * @author <a href="mailto:liaochuntao@live.com">liaochuntao</a>
+ */
+public class NacosAutoRefreshPropertySourceLoader implements PropertySourceLoader {
+    
+    private final Map<String, Object> properties = new ConcurrentHashMap<>(16);
+    
+    private Resource holder = null;
+    
+    @Override
+    public String[] getFileExtensions() {
+        return new String[] {"properties"};
+    }
+    
+    @Override
+    public List<PropertySource<?>> load(String name, Resource resource) throws IOException {
+        holder = resource;
+        Map<String, ?> tmp = loadProperties(resource);
+        properties.putAll(tmp);
+    
+        try {
+            WatchFileCenter.registerWatcher(ApplicationUtils.getConfFilePath(), new FileWatcher() {
+                @Override
+                public void onChange(FileChangeEvent event) {
+                    try {
+                        Map<String, ?> tmp1 = loadProperties(holder);
+                        properties.putAll(tmp1);
+                    } catch (IOException ignore) {
+                    
+                    }
+                }
+            
+                @Override
+                public boolean interest(String context) {
+                    return StringUtils.contains(context, "application.properties");
+                }
+            });
+        } catch (NacosException ignore) {
+        
+        }
+    
+        if (properties.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return Collections
+                .singletonList(new OriginTrackedMapPropertySource("nacos_application_conf", properties));
+    }
+    
+    private Map<String, ?> loadProperties(Resource resource) throws IOException {
+        return new OriginTrackedPropertiesLoader(resource).load();
+    }
+    
+    @JustForTest
+    protected Map<String, Object> getProperties() {
+        return properties;
+    }
+    
+    static class OriginTrackedPropertiesLoader {
```

Comment:
I found this inner class only used in current class line 97. And each time create a new one. Can this inner class refactor to some private method.

If there are other same package class need this class. Whether extract as an single class will be better?